astroid/nodes/node_ng.py

```diff
@@ -32,7 +32,7 @@
     from astroid import nodes
 
 if sys.version_info >= (3, 6, 2):
-    from typing import NoReturn
+    from typing import NoReturn  # pylint: disable=unused-import
```

Comment:
Should we move the import in the `if TYPE_CHECKING` block instead?

Reply:
Doesn't work sadly. The `TYPE_CHECKING` check for `unused-import` only checks one level. I'll investigate and open an issue/create a fix.

Reply:
You could remove the `sys.version_info` check then. We likely won't run `mypy` with Python < `3.6.2` anyway. At runtime `TYPE_CHECKING` is always `False`.

Reply:
But yeah, maybe we'll need to update the detection logic in `pylint`.

Reply:
> We likely won't run mypy with Python < 3.6.2

Should we change the required python version like in pylint ?

Reply:
> > We likely won't run mypy with Python < 3.6.2
> 
> Should we change the required python version like in pylint ?

Isn't that already planned (on some todo list)?

Reply:
Yes, I also realized I released 2.8.5 without taking care of that.

Reply:
Tracked this here now : https://github.com/PyCQA/pylint/issues/5250